extension/src/server/serverUtil.ts

```diff
@@ -15,13 +15,15 @@ export interface ProcessEnv {
     [key: string]: string | undefined;
 }
 
-export function getGradleServerEnv(): ProcessEnv {
-    const javaHome = getConfigGradleJavaHome();
+export async function getGradleServerEnv(): Promise<ProcessEnv | undefined> {
+    const javaHome = getJavaHome();
     const env = { ...process.env };
     if (javaHome) {
         Object.assign(env, {
             VSCODE_JAVA_HOME: javaHome,
         });
+    } else if (!(await checkEnvJavaExecutable())) {
+        return undefined;
```

Comment:
what will happen if `undefined` is returned? compared with previous behavior where `env` is returned.

Reply:
prompt NO_JAVA_EXECUTABLE early before actually executing it?

Reply:
> prompt NO_JAVA_EXECUTABLE early before actually executing it?

Yes. Please refer: https://github.com/microsoft/vscode-gradle/pull/1263/files#diff-3369ade313508c278005bfe83330cb8fb4852a75783f5f5843d8d89a46423fc6R43-R46, if we can find a proper java executable, will return early.

The previous behavior is always to run the start script via `cmd` variable, in this way, Gradle will report an error if no java executable can be found.